src/NServiceBus.Metrics/RawData/LongValueWriter.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace NServiceBus.Metrics.RawData
+{
+    using System;
+    using System.IO;
+
+    static class LongValueWriter
+    {
+        const long Version = 1;
+
+        // WIRE FORMAT, Version: 1
+
+        // 0                   1                   2                   3
+        // 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+        //+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        //|     Version   | Min date time | date1 |    value 1    | date2 |  
+        //+---------------+---------------+-------------------------------+
+        //|     value 2   | date3 |   value  3    | date4 |     value 4   |
+        //+---------------+---------------+-------------------------------+
+
+        public static void Write(BinaryWriter outputWriter, ArraySegment<RingBuffer.Entry> chunk)
```

Comment:
> What is an advantage of `ArraySegment`?

Over what? :) I assume `over an array`

ArraySegment is used to enable batching writes, it provides no-alloc, zero-copy way to pass all the values to writer in one go. My assumption was that copying lots of memory, when this can be avoided just by using a segment, is an easy win, reducing the overhead of reporter.